HIP/hip-0000-token-update-remove-keys.md

```diff
@@ -0,0 +1,196 @@
+- As a creator I want the flexibility to remove keys as my project evolves. For example, I might start out with KYC as a requirement and later decide that it is not necessary.
+- As an NFT minting service I want to be able to mint an NFT collection on behalf of a creator using our private key and then update the treasury account to the creator's account whilst simultaneously removing the admin key so the creator ends up with an immutable NFT collection in their treasury account.
+- As an author of a smart contract that I am now happy is operating smoothly, I would like to remove the admin key
+
+## Specification
+
+1. Introduce a constant to represent an empty key. This will be used as part of the token and/or contract update calls to remove existing keys.
+
+2. Update all other areas of the SDK that make use of an empty key to use this new constant for consistency.
+
+We use the JS SDK for demonstration using the HTS token entity as an example. The same approaches could be take for the `ContractUpdateTransaction`.
+
+### Extend `TokenUpdateTransaction` to Support Removing Keys
+
+All these keys should be removable as part of a `TokenUpdateTransaction` if they are present on a Token:
+
+- Admin Key
+- Wipe Key
+- KYC Key
+- Freeze Key
+- Pause Key
+- Supply Key
+- Fee Schedule Key
+
+**An example `TokenUpdateTransaction` to remove all keys from a Token.**
+
+```js
+let transaction = new TokenUpdateTransaction({
+  tokenId: "0.0.123456",
+  kycKey: Key.None,
+  freezeKey: Key.None,
+  pauseKey: Key.None,
+  wipeKey: Key.None,
+  supplyKey: Key.None,
+  feeScheduleKey: Key.None,
+  adminKey: Key.None,
+}).freezeWithClient(client);
+
+// Sign the transaction with the admin key
+const signTx = await transaction.sign(adminKey);
+```
+
+**Key.None**
+We can’t use `null` as the value for these fields because behind the scenes protobuf removes them for optimisation. An empty KeyList could be used however, since this action can't be undone, we propose having a dedicated constant `Key.None`. The constant provides clearer intent which is crucial when dealing with a transaction that can't be reversed.
+
+**Requirements**
+
+- If an empty KeyList is passed in, the SDK should throw an error to avoid accidental removal of keys.
+- The `TokenUpdateTransaction` should be able to remove any/all keys from a token if the key exists on the token.
+- If the Admin key is removed as part of the update transaction, then all other updates should happen first and the admin key removed last to avoid any `TOKEN_IS_IMMUTABLE` errors.
+- If a key doesn't exist on the token and a call is made to remove it then return a `TOKEN_HAS_NO_SUPPLY_KEY`, `TOKEN_HAS_NO_PAUSE_KEY` etc. This error response is the same as when trying to update a key that doesn't exist.
+- All transactions to remove a key must be signed with the admin key so if a token has no admin key, then any other keys present can't be removed.
```

Comment:
All transactions should be signed with all keys involved in the change, including those being removed and those being added _in addition to_ the ADMIN key.  This would be consistent with pre-existing HAPI behavior implemented elsewhere in the system.

Reply:
coming back to this...i agree that the existing key should not be required to sign. What if the owner no longer has access to it, but still has the ADMIN key? I believe the ADMIN key is currently the only one that needs to be signed with both existing and new keys.